fuse_models/test/test_graph_ignition.cpp

```diff
@@ -0,0 +1,411 @@
+  constraint1->data = 1.5;
+  graph.addConstraint(constraint1);
+
+  auto constraint2 = ExampleConstraint::make_shared(
+      "test",
+      std::initializer_list<fuse_core::UUID>{ variable2->uuid(), variable3->uuid() });  // NOLINT(whitespace/braces)
+  constraint2->data = -3.7;
+  graph.addConstraint(constraint2);
+
+  // Call the SetGraph service
+  fuse_models::SetGraph srv;
+  fuse_core::serializeGraph(graph, srv.request.graph);
+  const bool success = ros::service::call("/set_graph", srv);
+  ASSERT_TRUE(success);
+  EXPECT_TRUE(srv.response.success);
+
+  // The ignition sensor should publish a transaction in response to the service call. Wait for the callback to fire.
+  auto status = callback_future.wait_for(std::chrono::seconds(5));
+  ASSERT_TRUE(status == std::future_status::ready);
+
+  // Check the transaction is equivalent to the graph, i.e. it has the same constraints and transactions
+  const auto transaction = callback_future.get();
+
+  ASSERT_EQ(boost::size(graph.getConstraints()), boost::size(transaction->addedConstraints()));
+  ASSERT_EQ(boost::size(graph.getVariables()), boost::size(transaction->addedVariables()));
+
+  // We cannot compare the constraints or variables of the graph with the added constraints or variables of the
+  // transaction with:
+  //
+  //    graph.getConstraints() == transaction->addedConstraints()
+  //
+  // and
+  //
+  //    graph.getVariables() == transaction->addedVariables()
+  //
+  // because the graph could stored the constraints and variables in unordered containers. Indeed, the
+  // fuse_graphs::HashGraph uses unordered containers for both the constraints and variables.
+  //
+  // So even if the added constraints and variables are stored in std::vector containers in the transaction, we cannot
+  // compare them with the straightforward approach mentioned above. Instead, we need to check that all added
+  // constraints and varaibles are in the graph, and check they are the same.
+  for (const auto& added_constraint : transaction->addedConstraints())
+  {
+    try
+    {
+      const auto& constraint = graph.getConstraint(added_constraint.uuid());
+
+      EXPECT_EQ(constraint, added_constraint) << failure_description;
+    }
+    catch (const std::out_of_range& ex)
+    {
+      ADD_FAILURE() << ex.what();
+    }
+  }
+
+  for (const auto& added_variable : transaction->addedVariables())
+  {
+    try
+    {
+      const auto& variable = graph.getVariable(added_variable.uuid());
+
+      EXPECT_EQ(variable, added_variable) << failure_description;
+    }
+    catch (const std::out_of_range& ex)
+    {
+      ADD_FAILURE() << ex.what();
+    }
+  }
+
+  // Since the variables in the graph do not have a stamp, the transaction should have a single involved stamp, equal to
+  // the transaction stamp, that should also be equal to the requested graph message stamp
+  ASSERT_EQ(1, boost::size(transaction->involvedStamps()));
```

Comment:
Build doesn't like this for Bionic builds, because it's comparing signed and unsigned values:

```
ASSERT_EQ(1u, boost::size(transaction->involvedStamps()));
```

Reply:
Thanks. Fixed.

If that was the only error, now it should pass.